src/main/java/it/pagopa/pm/gateway/client/bpay/BancomatPayClient.java

```diff
@@ -1,45 +1,83 @@
 package it.pagopa.pm.gateway.client.bpay;
 
 import it.pagopa.pm.gateway.client.bpay.generated.*;
-import it.pagopa.pm.gateway.dto.*;
+import it.pagopa.pm.gateway.dto.BPayPaymentRequest;
+import it.pagopa.pm.gateway.dto.BPayRefundRequest;
+import it.pagopa.pm.gateway.exception.ExceptionsEnum;
+import it.pagopa.pm.gateway.exception.RestApiException;
 import it.pagopa.pm.gateway.utils.ClientUtils;
+import lombok.Data;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.aspectj.lang.annotation.Before;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
 import org.springframework.ws.client.core.WebServiceTemplate;
 
-import javax.xml.bind.*;
+import javax.annotation.PostConstruct;
+import javax.xml.bind.JAXBElement;
 import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static it.pagopa.pm.gateway.constant.ClientConfigs.*;
+
 
 @Slf4j
 public class BancomatPayClient {
 
     @Autowired
     private WebServiceTemplate webServiceTemplate;
 
-    @Value("${bancomatPay.client.group.code}")
-    public String GROUP_CODE;
-    @Value("${bancomatPay.client.institute.code}")
-    public String INSTITUTE_CODE;
-    @Value("${bancomatPay.client.tag}")
-    public String TAG;
-    @Value("${bancomatPay.client.token}")
-    public String TOKEN;
+    @Autowired
+    private Environment environment;
 
     private final ObjectFactory objectFactory = new ObjectFactory();
 
-    public InserimentoRichiestaPagamentoPagoPaResponse sendPaymentRequest(BPayPaymentRequest request, String guid) {
+    @Value("${bancomatPay.client.config}")
+    private String BANCOMAT_CLIENT_CONFIG;
+
+    private Map<String, String> configValues;
+
+    private void initConfigValues() throws RestApiException
+    {
+        if (!MapUtils.isEmpty(configValues)){
```

Comment:
Map is initialized every time a public method is called, but only if it has not yet been initialized.
This kind of init is necessary for Junit test, otherwise BANCOMAT_CLIENT_CONFIG is always null when the test runs.